File: Easy/0028-Implement strStr()/Linear Scan.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0
        for i in range(len(haystack) - len(needle) + 1):
            flag = True
            for j in range(0, len(needle)):
                if haystack[i + j] != needle[j]:
                    flag = False
                    break
            if flag:
                return i
        return -1

File: Problemset/0896-Monotonic Array/Linear Scan.rs
impl Solution {
    pub fn is_monotonic(a: Vec<i32>) -> bool {
        let mut flag = 0;

        for i in 1..a.len() {
            if a[i - 1] != a[i] {
                if flag * (a[i - 1] - a[i]) >= 0 {
                    flag = a[i - 1] - a[i];
                } else {
                    return false;
                }
            }
        }

        true
    }
}
